src/main/java/com/hivemq/client/internal/mqtt/handler/publish/incoming/MqttIncomingPublishFlow.java

```diff
@@ -92,7 +92,9 @@ public void onComplete() {
     @Override
     public void onError(final @NotNull Throwable t) {
         if (done) {
-            RxJavaPlugins.onError(t);
+            if (t != error) {
+                RxJavaPlugins.onError(t);
+            }
```

Comment:
What if this scenario occurs:
```
onError(throwable1)
onError(throwable2)
onError(throwable1)
onError(throwable2)
```

Reply:
For clarification: if a flow is registered for more than 1 subscription and all subscriptions are errored this method is called multiple times with the same throwable object.
Therefore a reference equality check is performed.
This check is actually just a safety check to detect bugs that accidentally call onError with different exceptions.
So if your described scenario occurs, there is a bug and `RxJavaPlugins.onError` is called rightly.